src/enkf/gridio_fv3reg.f90

```diff
@@ -541,20 +545,20 @@ subroutine readgriddata(nanal1,nanal2,vars3d,vars2d,n3d,n2d,levels,ndim,ntimes,f
          if(allocated(qsatworkvar3d)) deallocate(qsatworkvar3d)
        endif
        if(l_fv3reg_filecombined) then
-         call nc_check( nf90_close(file_id),&
-              myname_,'close '//trim(filename) )
+          call nc_check( nf90_close(file_id),&
+            myname_,'close '//trim(filename) )
        else
          do ifile=1,2
-            file_id=fv3lamfile%fv3lam_fileid(ifile)
-            filename=fv3lamfile%fv3lamfilename(ifile)
-            call nc_check( nf90_close(file_id),&
-            myname_,'close '//trim(filename) )
+           file_id=fv3lamfile%fv3lam_fileid(ifile)
+           filename=fv3lamfile%fv3lamfilename(ifile)
+           call nc_check( nf90_close(file_id),&
+             myname_,'close '//trim(filename) )
          enddo
        endif 
-       !======================================================================
-       ! Deallocate memory 
-       if(allocated(workvar3d))             deallocate(workvar3d)
-      end do ! ntile loop
+      !======================================================================
+      ! Deallocate memory 
+      if(allocated(workvar3d))             deallocate(workvar3d)
+    end do ! ntile loop
 
     end do backgroundloop ! loop over backgrounds to read in
     end do ensmemloop ! loop over ens members to read in 
```

Comment:
Please ensure that the backgroundloop and ensemloop do loops are properly indented through out.